examples/helloworld/src/index.js

```diff
@@ -1,4 +1,26 @@
 import React from 'react';
+import _ from 'underscore';
 import App from './App';
 
-React.render(<App />, document.getElementById('root'))
+
+var state = {
+    very: {
+        deeply: {
+            nested: {
+                counts: _.range(400).map(function () { return 0; })
+            }
+        }
+    }
+};
+
+function swapper(f) {
+    state = f(state);
```

Comment:
Yes, it will. but expose such action to user is not a good idea. IMO, at least abstract it inside a function.

In regards to mutation detection, I think @danielmiladinov 's solution is on the right way. #56